app/scripts/controllers/network-order.ts

```diff
@@ -289,7 +294,14 @@ export class NetworkOrderController extends BaseController<
       !chainIds.includes(selectedNetworkChainId) &&
       clientId
     ) {
-      this.messagingSystem.call('NetworkController:setActiveNetwork', clientId);
+      // Settimout delay to run this in a seperate 'tick'.
+      // There were some issues related to background state being updated, but persisted state not being updated.
+      setTimeout(() => {
+        this.messagingSystem.call(
+          'NetworkController:setActiveNetwork',
+          clientId,
+        );
+      }, 0);
```

Comment:
I'm still not entirely sure about why this does not correctly update the persisted state.
Maybe since this messenger call is `async` but the function is `sync`?